Substrate bridge documentation (#44)
sora-xor/sora-docs: src/establishing-hrmp-channel.md

> # Establish HRMP channel

Reviewer: ```suggestion
# Establish an HRMP channel
```

> To establish communication between your parachain and SORA, you must create HRMP channels that act as bridges for messages. This guide will walk you through the process using information from the official [Substrate documentation](https://docs.substrate.io/reference/how-to-guides/parachains/add-hrmp-channels/) but applied to SORA.

Reviewer 2: _"channels that act as bridges"_ is confusing and inaccurate. 
HRMP channels are channels that go through the Relay Chain to transfer messages between parachains.

Author: Solved by https://github.com/sora-xor/sora-docs/pull/44/files/52b12d9df8ed71b9b6fcc863710e7d7fd9496506#r1349775025

> To establish communication between your parachain and SORA, you must create HRMP channels that act as bridges for messages. This guide will walk you through the process using information from the official [Substrate documentation](https://docs.substrate.io/reference/how-to-guides/parachains/add-hrmp-channels/) but applied to SORA.

Reviewer 2: ```suggestion
To establish communication between your parachain and SORA, you must create HRMP channels that go through the Relay Chain to transfer messages between parachains.
```

> To establish communication between your parachain and SORA, you must create HRMP channels that act as bridges for messages. This guide will walk you through the process using information from the official [Substrate documentation](https://docs.substrate.io/reference/how-to-guides/parachains/add-hrmp-channels/) but applied to SORA.

Reviewer 2: ```suggestion
To establish communication between your parachain and SORA, you must create HRMP channels that go from on parachain to another through the Relay Chain to transfer messages between parachains. This guide will walk you through the process using information from the official [Substrate documentation](https://docs.substrate.io/reference/how-to-guides/parachains/add-hrmp-channels/) but applied to SORA.
```

:::

Once the extrinsic executes successfully, SORA needs to accept the channel establishment.

3. Subsequently, the SORA team will accept the channel by incorporating the encoded call data of `hrmp.hrmpAcceptOpenChannel`. This data will feature your parachain's ID as a parameter and will be executed via the `xcmApp.sudoSendXCM` call.

Upon successful execution, (an) HRMP channel(s) will be established from your parachain to SORA.

## SORA to Your Parachain

The steps here mirror those from the preceding sub-section. For steps (1) and (2) outlined earlier, SORA initiates an XCM message to propose channel establishment. Subsequently, it's your responsibility to encode `hrmp.hrmpAcceptOpenChannel` with SORA's parachain ID. For instance, in the case of Rococo, the encoded call data would be: `0x3c01db070000`. Execute an XCM message from your parachain in alignment with the [official guidelines](https://docs.substrate.io/reference/how-to-guides/parachains/add-hrmp-channels/), including the encoded call data containing SORA's parachain ID.

> Upon successful execution of this process, your parachain will be fully equipped to receive messages from the SORA parachain.

Reviewer 2: ```suggestion
Upon successful execution of this process, your parachain will be set up to receive messages from the SORA parachain.

We will still need to register the assets to be transferred from one chain to another. On your parachain, you will need to register the SORA assets. On the SORA Mainnet, we will need to register your assets. 
```

Reviewer 2: Need to add a small section with the info we need to register parachain assets on the SORA Mainnet and redirect to the SORA asset details to enable parachain teams to register SORA assets from this guide.

# Establish an HRMP channel

To establish communication between your parachain and SORA, you must create HRMP channels that go through the Relay Chain to transfer messages between parachains.

> Remember that HRMP channels usually go in one direction. For two-way communication, you'll need to set up another channel going the opposite way. Therefore, the following is split in to sub-sections:

Reviewer 3: ```suggestion
Remember that HRMP channels usually go in one direction. For two-way communication, you'll need to set up another channel going the opposite way. Therefore, the following is split into sub-sections:
```

## Your Parachain to SORA

1. Obtain the encoded call data of the `hrmp.hrmpInitOpenChannel` extrinsic on the relay chain.

> This unique code includes a `recipient` detail telling the relay chain which parachain you want to connect to. This detail changes depending on what environment you are planning to establish the channel in. For Kusama, look at [this list](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fksm-rpc.stakeworld.io#/parachains), and for Rococo, go to [this list](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-rpc.polkadot.io#/parachains).

Reviewer 3: ```suggestion
This unique code includes a `recipient` detail telling the relay chain which parachain you want to connect to. The recipient changes depending on what environment you are planning to establish the channel in. For Kusama, look at [this list](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fksm-rpc.stakeworld.io#/parachains), and for Rococo, go to [this list](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-rpc.polkadot.io#/parachains).
```

::: info

Combine this parachain id with the other info from the documentation, and you'll end up with something like this for Rococo: `0x3c00db070000e803000000900100`.

> :::

Reviewer 3: I don't think this should be in a note.

```suggestion
Combine this parachain id with the other info from the documentation. For example, for Rococo you'll have something like this: `0x3c00db070000e803000000900100`.
```
